radish/extensions/cucumber_json_writer.py

```diff
@@ -44,7 +44,14 @@ def generate_ccjson(self, features, marker):
         duration = timedelta()
         for feature in features:
             if feature.state in [Step.State.PASSED, Step.State.FAILED]:
-                duration += feature.duration
+                # feature file run started
+                if feature.starttime is not None:
+                    # feature file run not finished
+                    if feature.endtime is None:
```

Comment:
When is this case? That the feature was started but not ended?

Reply:
In my testing environment I want to preview tests results more dynamically - then just after all tests execution finish.

To manage that I am adding hook to generate_ccjson for all features that have finished and started after each step execution

And since there is feature is not finished yet - the endtime is set to None

It is really useful with https://github.com/damianszczepanik/cucumber-sandwich tool which generate report after each json change.

Reply:
Alright! Sounds reasonable to me!